// @polygon.io/client-js/lib/rest/indices/index.js
import { getWithGlobals } from "../transport/request.js";
import { aggregates } from "../stocks/aggregates.js";
import { dailyOpenClose, } from "../stocks/dailyOpenClose.js";
import { previousClose, } from "../stocks/previousClose.js";
import { snapshotTicker } from "../indices/snapshots.js";
import { sma } from "../stocks/sma.js";
import { ema } from "../stocks/ema.js";
import { macd } from "../stocks/macd.js";
import { rsi } from "../stocks/rsi.js";
export const indicesClient = (apiKey, apiBase = "https://api.polygon.io", options) => {
    const get = getWithGlobals(apiKey, apiBase, options);
    return ({
        aggregates: (...args) => aggregates(get, ...args),
        dailyOpenClose: (...args) => dailyOpenClose(get, ...args),
        previousClose: (...args) => previousClose(get, ...args),
        snapshotTicker: (...args) => snapshotTicker(get, ...args),
        sma: (...args) => sma(get, ...args),
        ema: (...args) => ema(get, ...args),
        macd: (...args) => macd(get, ...args),
        rsi: (...args) => rsi(get, ...args),
    });
};
export default indicesClient;
