// @polygon.io/client-js/lib/rest/options/index.js
import { getWithGlobals } from "../transport/request.js";
import { aggregates } from "./aggregates.js";
import { dailyOpenClose, } from "./dailyOpenClose.js";
import { previousClose } from "./previousClose.js";
import { trades } from "./trades.js";
import { lastTrade } from "./lastTrade.js";
import { quotes } from "./quotes.js";
import { snapshotOptionContract, snapshotOptionChain } from "./snapshots.js";
import { summaries } from "./summaries.js";
import { sma } from "./sma.js";
import { ema } from "./ema.js";
import { macd } from "./macd.js";
import { rsi } from "./rsi.js";
export { snapshotOptionContract, snapshotOptionChain, } from "./snapshots.js";
export const optionsClient = (apiKey, apiBase = "https://api.polygon.io", options) => {
    const get = getWithGlobals(apiKey, apiBase, options);
    return ({
        aggregates: (...args) => aggregates(get, ...args),
        summaries: (...args) => summaries(get, ...args),
        dailyOpenClose: (...args) => dailyOpenClose(get, ...args),
        previousClose: (...args) => previousClose(get, ...args),
        trades: (...args) => trades(get, ...args),
        lastTrade: (...args) => lastTrade(get, ...args),
        quotes: (...args) => quotes(get, ...args),
        snapshotOptionContract: (...args) => snapshotOptionContract(get, ...args),
        snapshotOptionChain: (...args) => snapshotOptionChain(get, ...args),
        sma: (...args) => sma(get, ...args),
        ema: (...args) => ema(get, ...args),
        macd: (...args) => macd(get, ...args),
        rsi: (...args) => rsi(get, ...args)
    });
};
export default optionsClient;
