// @polygon.io/client-js/lib/rest/reference/index.js
import { getWithGlobals } from "../transport/request.js";
import { conditions } from "./conditions.js";
import { exchanges } from "./exchanges.js";
import { marketHolidays } from "./marketHolidays.js";
import { marketStatus } from "./marketStatus.js";
import { optionsContract, } from "./optionsContract.js";
import { optionsContracts, } from "./optionsContracts.js";
import { stockDividends } from "./dividends.js";
import { stockSplits } from "./stockSplits.js";
import { stockFinancials } from "./stockFinancials.js";
import { tickerDetails } from "./tickerDetails.js";
import { tickerNews } from "./tickerNews.js";
import { tickers } from "./tickers.js";
import { tickerTypes } from "./tickerTypes.js";
export const referenceClient = (apiKey, apiBase = "https://api.polygon.io", options) => {
    const get = getWithGlobals(apiKey, apiBase, options);
    return ({
        conditions: (...args) => conditions(get, ...args),
        exchanges: (...args) => exchanges(get, ...args),
        marketHolidays: (...args) => marketHolidays(get, ...args),
        marketStatus: (...args) => marketStatus(get, ...args),
        optionsContract: (...args) => optionsContract(get, ...args),
        optionsContracts: (...args) => optionsContracts(get, ...args),
        dividends: (...args) => stockDividends(get, ...args),
        stockSplits: (...args) => stockSplits(get, ...args),
        stockFinancials: (...args) => stockFinancials(get, ...args),
        tickerDetails: (...args) => tickerDetails(get, ...args),
        tickerNews: (...args) => tickerNews(get, ...args),
        tickers: (...args) => tickers(get, ...args),
        tickerTypes: (...args) => tickerTypes(get, ...args),
    });
};
export default referenceClient;
