// @polygon.io/client-js/lib/rest/stocks/index.js
import { getWithGlobals } from "../transport/request.js";
import { aggregates } from "./aggregates.js";
import { aggregatesGroupedDaily, } from "./aggregatesGroupedDaily.js";
import { dailyOpenClose, } from "./dailyOpenClose.js";
import { previousClose, } from "./previousClose.js";
import { lastQuote } from "./lastQuote.js";
import { lastTrade } from "./lastTrade.js";
import { snapshotAllTickers, snapshotGainersLosers, snapshotTicker, } from "./snapshots.js";
import { quotes } from "./quotes.js";
import { trades } from "./trades.js";
import { summaries } from "./summaries.js";
import { sma } from "./sma.js";
import { ema } from "./ema.js";
import { macd } from "./macd.js";
import { rsi } from "./rsi.js";
export const stocksClient = (apiKey, apiBase = "https://api.polygon.io", options) => {
    const get = getWithGlobals(apiKey, apiBase, options);
    return ({
        aggregates: (...args) => aggregates(get, ...args),
        aggregatesGroupedDaily: (...args) => aggregatesGroupedDaily(get, ...args),
        summaries: (...args) => summaries(get, ...args),
        dailyOpenClose: (...args) => dailyOpenClose(get, ...args),
        lastQuote: (...args) => lastQuote(get, ...args),
        lastTrade: (...args) => lastTrade(get, ...args),
        previousClose: (...args) => previousClose(get, ...args),
        quotes: (...args) => quotes(get, ...args),
        snapshotAllTickers: (...args) => snapshotAllTickers(get, ...args),
        snapshotGainersLosers: (...args) => snapshotGainersLosers(get, ...args),
        snapshotTicker: (...args) => snapshotTicker(get, ...args),
        sma: (...args) => sma(get, ...args),
        ema: (...args) => ema(get, ...args),
        macd: (...args) => macd(get, ...args),
        rsi: (...args) => rsi(get, ...args),
        trades: (...args) => trades(get, ...args)
    });
};
export default stocksClient;
